src/dsync/dsync.c

```diff
@@ -2384,6 +2901,26 @@ int main(int argc, char **argv)
     mfu_flist flist1 = mfu_flist_new();
     mfu_flist flist2 = mfu_flist_new();
 
+    mfu_param_path* link_dest_path = NULL;
+    mfu_flist flist_link_dest = NULL;     
+
+    if (options.link_dest != NULL) {
+        rc = dsync_validate_link_dest(options.link_dest, destpath->path);
+        if (rc < 0) {
+             /* disable link_dest if not avliable */
+            if (rank == 0) {
+                MFU_LOG(MFU_LOG_ERR, "Invalid link_dest: [%s], ret:%d. Disabled link_dest",
+                        options.link_dest, rc);
+            }
+            mfu_free(&options.link_dest);
+            options.link_dest = NULL;
```

Comment:
mfu_free will also set the pointer back to NULL, so let's remove this line.

Reply:
Fixed it.